test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+    (torch.triu_indices, lambda row, col, offset=0, dtype=torch.long, device='cpu', layout=torch.strided: -1),
+    (torch.trunc, lambda input, out=None: -1),
+    (torch.unbind, lambda input, dim=0: -1),
+    (torch.unique, lambda input, sorted=True, return_inverse=False, return_counts=False, dim=None: -1),
+    (torch.unique_consecutive, lambda input, return_inverse=False, return_counts=False, dim=None: -1),
+    (torch.unsqueeze, lambda input, dim, out=None: -1),
+    (torch.var, lambda input: -1),
+    (torch.var_mean, lambda input: -1),
+    (torch.where, lambda condition, x, y: -1),
+    (torch.zeros_like, lambda input, dtype=None, layout=None, device=None, requires_grad=False: -1),
+)
+
+TENSOR_LIKE_OVERRIDES = tuple(t[0] for t in TENSOR_LIKE_TORCH_IMPLEMENTATIONS)
+
+# TODO make these functions overridable
+TENSOR_LIKE_SKIP_TESTS = (
+    torch.sparse_coo_tensor,
+    torch.numel,
+)
+
+def generate_tensor_like_torch_implementations():
+    torch_vars = vars(torch)
+    untested_funcs = []
+    for func_name in torch.__all__ + dir(torch._C._VariableFunctions):
+        # these are functions that are public but shouldn't be overriden by __torch_function__
+        if func_name in IGNORED_TORCH_FUNCTIONS:
+            continue
+        # ignore private functions
+        if func_name.startswith('_'):
+            continue
+        # ignore in-place operators
+        if func_name.endswith('_'):
+            continue
+        # only consider objects with lowercase names
+        if not func_name.islower():
+            continue
+        func = torch_vars[func_name]
+        if func not in TENSOR_LIKE_OVERRIDES:
+            untested_funcs.append("torch.{}".format(func.__name__))
+    msg = (
+        "The following functions are not tested for __torch_function__ "
+        "support, please either add an entry in "
+        "TENSOR_LIKE_TORCH_IMPLEMENTATIONS for this function or if a "
+        "__torch_function__ override does not make sense, add an entry to "
+        "IGNORED_TORCH_FUNCTIONS.\n\n{}"
+    )
+    assert len(untested_funcs) == 0, msg.format(pprint.pformat(untested_funcs))
+    return TENSOR_LIKE_TORCH_IMPLEMENTATIONS
+
+@six.add_metaclass(ImplementationMeta)
+class TensorLike(object):
+    """A class that emulate the full tensor API
+
+    This class is used to explicitly test that the full torch.tensor API
+    can be overriden with a class that defines __torch_function__.
+    """
+    TORCH_IMPLEMENTATIONS = generate_tensor_like_torch_implementations()
+    IMPLEMENTS_DECORATOR = implements_tensor_like
+
+    def __torch_function__(self, func, args=(), kwargs=None):
+        if(kwargs is None):
+            kwargs = {}
+
+        if func not in HANDLED_FUNCTIONS_TENSOR_LIKE:
+            return NotImplemented
+        # In this case _torch_function_ should override TensorLike objects
+        return HANDLED_FUNCTIONS_TENSOR_LIKE[func](*args, **kwargs)
+
+class TestApis(TestCase):
+    def run(self, result=None):
```

Comment:
Do we really want to special-case how the tests are run? There are options that can be passed to unittest and pytest that would cause them to stop after the first error.

Reply:
Oh no, we don't, this should be removed. I added this for debugging, I didn't realize there was an option for UnitTest to stop on the first error.